MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm

```diff
@@ -1446,22 +1446,20 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
     // The depth value (including vertex position Z value) is held in the last index.
     clearColors[kMVKClearAttachmentDepthStencilIndex] = { _mtlDepthVal, _mtlDepthVal, _mtlDepthVal, _mtlDepthVal };
 
-    VkFormat vkAttFmt = subpass->getDepthStencilFormat();
-	MTLPixelFormat mtlAttFmt = pixFmts->getMTLPixelFormat(vkAttFmt);
-    _rpsKey.attachmentMTLPixelFormats[kMVKClearAttachmentDepthStencilIndex] = mtlAttFmt;
-
-	bool isClearingDepth = _isClearingDepth && pixFmts->isDepthFormat(mtlAttFmt);
-	bool isClearingStencil = _isClearingStencil && pixFmts->isStencilFormat(mtlAttFmt);
-    if (!isClearingDepth && !isClearingStencil) {
-        // If the subpass attachment isn't actually used, don't try to clear it.
-        _rpsKey.disableAttachment(kMVKClearAttachmentDepthStencilIndex);
-    }
-
-	if (!_rpsKey.isAnyAttachmentEnabled()) {
-		// Nothing to do.
-		return;
+	MTLPixelFormat mtlDSFmt = pixFmts->getMTLPixelFormat(subpass->isStencilAttachmentUsed()
+														 ? subpass->getStencilFormat()
+														 : subpass->getDepthFormat());
+	_rpsKey.attachmentMTLPixelFormats[kMVKClearAttachmentDepthStencilIndex] = mtlDSFmt;
+
+	// If neither the depth or stencil attachments are being cleared, nor being used, don't try to clear them.
+	bool isClearingDepth = _isClearingDepth && subpass->isDepthAttachmentUsed();
+	bool isClearingStencil = _isClearingStencil && subpass->isStencilAttachmentUsed();
+	if ( !isClearingDepth && !isClearingStencil ) {
+		_rpsKey.disableAttachment(kMVKClearAttachmentDepthStencilIndex);
 	}
```

Comment:
Thanks for prodding me on this one. We should indeed have an internally consistent approach to separating D & S attachments.

Clearing was already working, because D/S is always the same image view in Vulkan. But we should make it consistent with the other re-alignments. I've realigned this into separate D & S settings now.